### ISSUE TITLE
wr.s3.to_parquet() fails to parse map<string,string>

### ISSUE THREAD

**Issue by pixelizedPeanut** [aws/aws-sdk-pandas#2217]
### Describe the bug

The input file is a json file with a field that can have multiple values. I want to transform it to parquet where a field will be of type map<string,string>. This always errors and it's complaining it's expecting a list. Makes little sense to me. Please help.

### How to Reproduce

```
import awswrangler as wr
import pandas as pd

df = pd.DataFrame([{
    "details": {
        "detail1": "value1",
        "detail2":"value2"
    }
}])

wr.s3.to_parquet(
    df=df,
    dataset=True,
    path="dummy-location",
    database="dummy-db",
    table="dummy-table",
    dtype={"details": "map<string, string>"},
)
```

### Error
```
Exception in *** :  ("Could not convert {'detail1': 'value1','detail2':'value2'} with type dict: was not a sequence or recognized null for conversion to list type", 'Conversion failed for column details with type object')
```

### Expected behavior

write parquet file successfully 

### OS

Linux

### Python version

3.9

### AWS SDK for pandas version

arn:aws:lambda:eu-west-1:336392948345:layer:AWSSDKPandas-Python39:3

**Comment by LeonLuttenberger** [comment]
Hey,

The version of the layer you are using is using AWS SDK for pandas version `2.19.0`, which is a few months old. I tested the code in the newest version (`3.0.0`), and the example ran successfully.

You can upgrade to the Layer ARN version `arn:aws:lambda:eu-west-1:336392948345:layer:AWSSDKPandas-Python38:6`, which includes the newest version of the library.

Can you upgrade your version and please confirm whether it works for you as well?

Best regards,
Leon

**Comment by pixelizedPeanut** [comment]
Any layer available on python 3.9? We have tried using that layer and still doesn't work, also we get a few other odd issues I assume to it looking for python 3.8. We have tried downgrading python to 3.8, but that didn't fix it either. We're looking into it still, will get back with more details

**Comment by kukushking** [comment]
Yes Python 3.9 is available, the correct ARN from the [table](https://aws-sdk-pandas.readthedocs.io/en/stable/layers.html) is: `arn:aws:lambda:eu-west-1:336392948345:layer:AWSSDKPandas-Python39:6`
